### ISSUE TITLE
Clock drift mitigation suggestion not working

### ISSUE THREAD

**Issue by joknu1** [instacart/truetime-android#121]
Under `Clock drift and the need for Re-syncing` the following is stated:
> rerun the TrueTime.init call at regular/frequent intervals so you resynchronize with the true time

However looking at the lib code gives:
```
    protected void initialize(String ntpHost) throws IOException {
        if (isInitialized()) {
            TrueLog.i(TAG, "---- TrueTime already initialized from previous boot/init");
            return;
        }
        requestTime(ntpHost);
        saveTrueTimeInfoToDisk();
    }
```
which effectively skips fetching new time from NTP if value already cached in RAM. (I'm not using saving to prefs option.)

So how can I make the lib sync time every now and then?

**Comment by Colanderr** [comment]
I came across the same issue, @joknu1 have you found a workaround for this?
My only idea is to clear the cache, but this way I lose the ability to check the time until a new initialization goes through successfully.

**Comment by joknu1** [comment]
> I came across the same issue, @joknu1 have you found a workaround for this?
> My only idea is to clear the cache, but this way I lose the ability to check the time until a new initialization goes through successfully.

No, I didn't. I still use this lib, but since the project I'm working on only cares for drift in the scale of 1 hour, it does not seem critical for our project's purpose. Le t me know if you find any workaround @Colanderr !

**Comment by john-stemberger** [comment]
I have created a kotlin extension as a work around for this issue if anyone is interested

https://gist.github.com/john-stemberger/bab63c3935cdda052d2baf92732a2b85

**Comment by Matt7170** [comment]
How do I register for True time as a new instacart shopper?

**Comment by stAyushJain** [comment]
@kaushikgopal Need your help. we are using vanilla version in our app and Re-sync seems impossible if using cache option is enabled. Which method should I use to do re-sync in vanilla version?. Because 'requestTime'  is a package-private in 'TrueTime'.

**Comment by liu0527aa** [comment]
> 如果有人感兴趣的话，我已经创建了一个 kotlin 扩展作为解决此问题的方法
> 
> https://gist.github.com/john-stemberger/bab63c3935cdda052d2baf92732a2b85

hello,i want to use you method. However, it can't be allowed to change this code as the library. Could you tell the solition

**Comment by stAyushJain** [comment]
@liu0527aa you can try alpha version of truetime. There you do not need to use `requestTime(ntpHost)`. Or else you can use mine approach where you need to use custom implementation of Cache and update key `KEY_CACHED_BOOT_TIME`  manually.

**Comment by liu0527aa** [comment]
> @liu0527aa you can try alpha version of truetime. There you do not need to use `requestTime(ntpHost)`. Or else you can use mine approach where you need to use custom implementation of Cache and update key `KEY_CACHED_BOOT_TIME` manually.

I feel very embarrassed to ask where can i find the alpha version of truetime。。。And how to use the alpha version

**Comment by stAyushJain** [comment]
4.0.0.alpha @liu0527aa 

<img width="430" alt="Screenshot 2023-08-10 at 5 41 59 PM" src="https://github.com/instacart/truetime-android/assets/17740844/2b970a2e-ee82-4b64-be19-7eaafa0db359">

**Comment by liu0527aa** [comment]
> 4.0.0.alpha@liu0527aa
> 
> <img alt="屏幕截图 2023-08-10 下午 5 41 59" width="430" src="https://user-images.githubusercontent.com/17740844/259736125-2b970a2e-ee82-4b64-be19-7eaafa0db359.png">

yes,i find it。however, when i use the library of 4.0.0. I met this mistake，the author didn't push the 4.0.0.alpha to jitpack.io。could u give me some suggestion? thanks!
![image](https://github.com/instacart/truetime-android/assets/55947582/ac17b472-eb72-48d6-b987-d693869b95a5)

**Comment by CyxouD** [comment]
At 3.5 version, I fix this by calling `.initializeNtp` instead of `.initializeRx`. `initializeNtp` is called internally by `.initializeRx` if `TrueTimeRx` isn't initialized. It helps to not use cached version during clock re-sync.
It looks like this 
```
TrueTimeRx.build()
                    .initializeNtp(NTP_APPLE)
                    .subscribeOn(Schedulers.io())
                    .map { TrueTimeRx.now() }
```

Methods implementation:
<img width="678" alt="image" src="https://github.com/instacart/truetime-android/assets/17162008/69ada8ef-8269-4be2-8152-ef8576316f00">
